generic-override/src/Data/Override/Internal.hs

```diff
@@ -86,15 +92,50 @@ instance (GOverride xs f, GOverride xs g) => GOverride xs (f :*: g) where
   overrideFrom (f :*: g) = overrideFrom @xs f :*: overrideFrom @xs g
   overrideTo (f :*: g) = overrideTo @xs f :*: overrideTo @xs g
 
-instance GOverride xs (M1 S ('MetaSel ms su ss ds) (K1 R c)) where
+instance (GOverride xs f, GOverride xs g) => GOverride xs (f :+: g) where
+  type OverrideRep xs (f :+: g) = OverrideRep xs f :+: OverrideRep xs g
+  overrideFrom (L1 l) = L1 (overrideFrom @xs l)
+  overrideFrom (R1 r) = R1 (overrideFrom @xs r)
+  overrideTo (L1 l) = L1 (overrideTo @xs l)
+  overrideTo (R1 r) = R1 (overrideTo @xs r)
+
+-- | Auxiliary family to put Overridden under Maybe.
+--
+-- This is required to preserve aeson's omitNothingFields behavior
+-- if the user has enabled it.
+type family OverrideField ms c xs where
+  OverrideField ms (Maybe c) xs = Maybe (Overridden ms c xs)
+  OverrideField ms c xs = Overridden ms c xs
```

Comment:
This feels a bit like the tail wagging the dog. Also, it's not exactly clear why this is necessary. Can you explain?

Reply:
Oh I see, aeson is looking specifically for `Maybe c` but we give it `Overridden ms (Maybe c) xs`.

I haven't tested to confirm, but another concern with this approach is that it seems like it would affect our ability to override an instance for `Maybe`, e.g. 

```haskell
Override Foo '[ Maybe Text `As` OptionalText ]
```

Reply:
> Oh I see, aeson is looking specifically for `Maybe c` but we give it `Overridden ms (Maybe c) xs`.

Yes, that's why I had to introduce this type family. And `DoOverrideField` type class is needed and has to be multi-param instead of type class with associated type, because otherwise GHC complains about overlapping instances.

> I haven't tested to confirm, but another concern with this approach is that it seems like it would affect our ability to override an instance for `Maybe`, e.g.
> 
> ```haskell
> Override Foo '[ Maybe Text `As` OptionalText ]
> ```

I haven't thought of that, will check. What is `OptionalText` supposed to do btw?

Reply:
I didn't really have anything in mind for OptionalText, it was just for the example.